### PR TITLE
Add support for testing sets in results for feature tests.

### PR DESCRIPTION
**Pull request by kenhuuu**
gremlin-javascript doesn't support sets so the tests are ignored for now.

This PR is needed in order to support the implemention of list functions that are part of Proposal 3.

### CODE REVIEW

File: gremlin-go/driver/cucumber/cucumberSteps_test.go [hunk 1]
```diff
@@ -652,6 +654,22 @@ func compareListEqualsWithoutOrder(expected []interface{}, actual []interface{})
 					break
 				}
 			}
+		} else if reflect.TypeOf(a).String() == "*gremlingo.SimpleSet" {
+			// Set is a special case here because there is no TypeOf().Kind() for sets.
+			actualStringArray := makeSortedStringArrayFromSet(a)
+
+			for i := len(expectedCopy) - 1; i >= 0; i-- {
+				curExpected := expectedCopy[i]
+				if curExpected != nil && reflect.TypeOf(curExpected).String() == "*gremlingo.SimpleSet" {
```
Comment (vkagamlyk) [inline gremlin-go/driver/cucumber/cucumberSteps_test.go:663]: nit: better to use type casting with error checking like in https://github.com/apache/tinkerpop/blob/master/gremlin-go/driver/result.go#L153

**Comment by vkagamlyk** [comment]
LGTM, VOTE+1

**Comment by Cole-Greer** [comment]
LGTM. Thanks Ken, VOTE +1

File: gremlin-go/driver/cucumber/cucumberSteps_test.go (hunk 1 above)
Comment (kenhuuu) [reply gremlin-go/driver/cucumber/cucumberSteps_test.go]: Updated to use type assertion as suggested.

**Comment by kenhuuu** [comment]
VOTE +1

**Pull request merged**